device-code-flow-console/PublicAppUsingDeviceCodeFlow.cs

```diff
@@ -0,0 +1,178 @@
+﻿using Microsoft.Identity.Client;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace device_code_flow_console
+{
+    /// <summary>
+    /// Base class for public client applications in the sample that leverage the DeviceCodeFlow to 
+    /// authenticate a user
+    /// </summary>
+    public class PublicAppUsingDeviceCodeFlow
+    {
+        /// <summary>
+        /// Constructor of a public application leveraging Device Code Flow to sign-in a user
+        /// </summary>
+        /// <param name="app">MSAL.NET Public client application</param>
+        /// <param name="httpClient">HttpClient used to call the protected Web API</param>
+        /// <remarks>
+        /// For more information see https://aka.ms/msal-net-device-code-flow
+        /// </remarks>
+        protected PublicAppUsingDeviceCodeFlow(PublicClientApplication app, HttpClient httpClient)
+        {
+            App = app;
+            Client = httpClient;
+        }
+        protected PublicClientApplication App { get; private set; }
+
+        protected HttpClient Client { get; private set; }
+
+        /// <summary>
+        /// Acquires a token from the token cache, or device code flow
+        /// </summary>
+        /// <returns>An AuthenticationResult if the user successfully signed-in, or otherwise <c>null</c></returns>
+        public async Task<AuthenticationResult> AcquireATokenFromCacheOrDeviceCodeFlow(IEnumerable<String> scopes)
+        {
+            AuthenticationResult result;
+            var accounts = await App.GetAccountsAsync();
+
+            if (accounts.Any())
+            {
+                try
+                {
+                    // Attempt to get a token from the cache (or refresh it silently if needed)
+                    result = await App.AcquireTokenSilentAsync(scopes, accounts.FirstOrDefault());
+                }
+                catch (MsalUiRequiredException)
+                {
+                    result = null;
+                }
+            }
+            else
+            {
+                result = null;
```

Comment:
initialize result with null and eliminate this else clause

Reply:
Done. Thanks!